src/utils/group.ts

```diff
@@ -20,51 +20,70 @@
  */
 
 import { assert } from "../util";
+import { View, ViewMode } from "../components/editor/View";
 import { File } from "../model";
 
 export default class Group {
-  file: File;
-  files: File[];
-  preview: File;
-  constructor(file: File, preview: File, files: File[]) {
-    this.file = file;
+  view: View;
+  views: View[];
+  preview: View;
+
+  constructor(view: View, views: View[] = [], preview?: View) {
+    this.view = view;
+    this.views = views;
     this.preview = preview;
-    this.files = files;
   }
-  open(file: File, shouldPreview: boolean = true) {
-    const files = this.files;
-    const index = files.indexOf(file);
+  open(view: View, shouldPreview = true) {
+    const views = this.views;
+    const index = views.indexOf(view);
+
     if (index >= 0) {
-      // Switch to file if it's already open.
-      this.file = file;
+      // Switch to the view if it's already open.
+      this.view = view;
       if (!shouldPreview) {
         this.preview = null;
       }
-      return;
+
+      return ;
     }
     if (shouldPreview) {
       if (this.preview) {
         // Replace preview file if there is one.
-        const previewIndex = files.indexOf(this.preview);
+        const previewIndex = views.indexOf(this.preview);
         assert(previewIndex >= 0);
-        this.file = this.preview = files[previewIndex] = file;
+        this.view = this.preview = views[previewIndex] = view;
       } else {
-        files.push(file);
-        this.file = this.preview = file;
+        views.push(view);
+        this.view = this.preview = view;
       }
     } else {
-      files.push(file);
-      this.file = file;
+      views.push(view);
+      this.view = view;
       this.preview = null;
     }
   }
-  close(file: File) {
-    const i = this.files.indexOf(file);
-    assert(i >= 0);
-    if (file === this.preview) {
-      this.preview = null;
+  openFile(file: File, preview = true, mode: ViewMode = ViewMode.EDITOR) {
+    const files = this.views.map(view => view.file);
```

Comment:
use `this.views.findIndex(view => view.file === file)` to find the `index`

Reply:
@mbebenita Right now, the only location we call openFile is the files tree, and it's in editor mode. So I left it as such in the hope I don't get too far changing everything as I usually do